src/modules/settings/components/Store.jsx

```diff
@@ -0,0 +1,68 @@
+import React, {useEffect, useState} from 'react';
+import settings from '../../../settings.js';
+
+export const Components = {};
+
+export function registerComponent(Component, metadata) {
+  Components[metadata.settingId] = {
+    ...metadata,
+    render: (...props) => <Component {...props} />,
+  };
+}
+
+export function useStorageState(settingId) {
+  const [value, setValue] = useState(settings.get(settingId));
+
+  useEffect(() => {
+    function callback(newValue) {
+      setValue(newValue);
+    }
+
+    settings.on(`changed.${settingId}`, callback);
+
+    return () => {
+      settings.off(`changed.${settingId}`, callback);
+    };
+  }, []);
+
+  function setSetting(newValue) {
+    if (newValue === value) return;
+    setValue(newValue);
+    settings.set(settingId, newValue);
+  }
+
+  return [value, setSetting];
+}
+
+// useStorageRadioState() - Will return the index of the first setting that is enabled otherwise -1
+export function useStorageRadioState(settingIds) {
```

Comment:
since radio storage is storing a singular value, I'm curious why we don't store an enum value instead?

```
const ChatLayouts = {
  LEFT: 1,
  RIGHT: 2
};

...

selected={value === ChatLayouts.LEFT}
```